opennlp-tools/src/test/java/opennlp/tools/eval/AbstractEvalTest.java

```diff
@@ -18,73 +18,65 @@
 package opennlp.tools.eval;
 
 import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
 import java.math.BigInteger;
-import java.nio.file.Files;
-import java.nio.file.Path;
+import java.nio.charset.StandardCharsets;
 import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
 
 import org.junit.Assert;
 
 import opennlp.tools.ml.maxent.quasinewton.QNTrainer;
 import opennlp.tools.ml.naivebayes.NaiveBayesTrainer;
 import opennlp.tools.ml.perceptron.PerceptronTrainer;
+import opennlp.tools.util.ObjectStream;
 import opennlp.tools.util.TrainingParameters;
 import opennlp.tools.util.model.ModelUtil;
 
-public class EvalUtil {
+public abstract class AbstractEvalTest {
 
-  static final double ACCURACY_DELTA = 0.0001d;
+  public static final double ACCURACY_DELTA = 0.0001d;
+  public static final String HASH_ALGORITHM = "MD5";
 
-  static TrainingParameters createPerceptronParams() {
+  public static void verifyTrainingData(ObjectStream<?> samples, BigInteger checksum) throws Exception {
+
+    MessageDigest digest = MessageDigest.getInstance(HASH_ALGORITHM);
+
+    Object sample;
+    while ((sample = samples.read()) != null) {
+      digest.update(sample.toString().getBytes(StandardCharsets.UTF_8));
+    }
+
+    samples.close();
+
+    Assert.assertEquals(checksum, new BigInteger(1, digest.digest()));
+
+  }
+
+  public static File getOpennlpDataDir() {
+    return new File(System.getProperty("OPENNLP_DATA_DIR"));
```

Comment:
This is not a nice way of having it fail. If the data dir is not set it throws a NullPointerException. I propose we catch that case and throw an exception with a proper description.